### PR TITLE
IBP-4064 import lot updates

### PR DESCRIPTION
**Pull request by cuenyad**
Hi @nahuel-soldevilla and @clarysabel 

Include:
https://github.com/IntegratedBreedingPlatform/InventoryManager/pull/62
https://github.com/IntegratedBreedingPlatform/Middleware/pull/893

Please review
Regards, Diego

### CODE REVIEW

File: src/main/java/org/ibp/api/rest/inventory/manager/LotResource.java [hunk 1]
```diff
@@ -198,12 +202,32 @@ public long getCount() {
...
+			final LotsSearchDto searchDTO =
+				this.searchRequestDtoResolver.getLotsSearchDto(lotRequest.getSingleInput().getSearchComposite());
+
+			extendedLotDtos = this.lotService.searchLots(searchDTO, null);
+			if (lotRequest.getSingleInput().getSearchComposite().getSearchRequest() == null) {
+				this.extendedLotListValidator
+					.validateAllProvidedLotUUIDsExist(extendedLotDtos, lotRequest.getSingleInput().getSearchComposite().getItemIds());
+			}
+		} else {
+			final List<String> lotUIDs =
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/inventory/manager/LotResource.java:225]: @cuenyad are we validating that the UIDs can't be repeated? How is the bahaviour in that case?

File: src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/validator/LotInputValidator.java [hunk 2]
```diff
@@ -133,7 +164,32 @@ private void validateTransactionStatus(final List<ExtendedLotDto> lotDtos, final
...
 		if (transactionDtos.stream().map(TransactionDto::getTransactionStatus)
 			.anyMatch(s -> s.equals(TransactionStatus.CONFIRMED.getValue()))
-			&& (updateRequestDto.getUnitId() != null)) {
+			&& (lotSingleUpdateRequestDto.getUnitId() != null)) {
+
+			this.errors.reject("lots.transactions.status.confirmed.cannot.change.unit");
+		}
+	}
+
+	private void validateTransactionStatus(final LotMultiUpdateRequestDto lotMultiUpdateRequestDto) {
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/validator/LotInputValidator.java:173]: I will try to make this validation more generic, with a parameter List<String> lotUUID and name validateNoConfirmedTransactions and unify both functions

File: src/main/java/org/ibp/api/java/impl/middleware/inventory/common/validator/InventoryCommonValidator.java [hunk 3]
```diff
@@ -82,4 +83,17 @@ public void validateLotNotes(final String notes, final BindingResult errors) {
 		}
 	}
 
+	public void validateLotNotes(final List<String> notes, final BindingResult errors) {
+		if (notes.stream().anyMatch(s -> StringUtils.isBlank(s))) {
+			errors.reject("lot.input.list.notes.null.or.empty", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+		}
+
+		final List<String> filteredNotes =
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/inventory/common/validator/InventoryCommonValidator.java:92]: This validation will be better implemented with findAny, we dont need in this case to know the number of invalid notes.

**Review by clarysabel — CHANGES_REQUESTED** [review]
Hi @cuenyad please review my comments in this PR

File: src/main/java/org/ibp/api/rest/inventory/manager/LotResource.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/rest/inventory/manager/LotResource.java:225]: @clarysabel I added a validateLotUUIDsDuplicated for this case.

**Pull request merged**